glommio/src/lib.rs

```diff
@@ -389,26 +406,28 @@ pub mod prelude {
 /// parameter consisting of the return type of the future encapsulated by this
 /// task.
 ///
-/// However for associated functions without a self parameter, like local() and
-/// local_into(), the type is always () and Rust is not able to elide.
+/// However for associated functions without a self parameter, like `local()` and
+/// `local_into()`, the type is always `()` and Rust is not able to elide.
 ///
-/// Writing Task::<()>::function() works, but it is not very ergonomic.
+/// Writing `Task::<()>::function()` works, but it is not very ergonomic.
 pub type Local = Task<()>;
 
-/// An attribute of a TaskQueue, passed during its creation.
+/// An attribute of a [`TaskQueue`], passed during its creation.
```

Comment:
Note that `TaskQueue` is not an exported type of this crate; only `TaskQueueHandle` is exported.

I wonder if we could perhaps change the interface a bit to have `spawn_into` be a method of `TaskQueueHandle`, and then rename the private `TaskQueue` to... I don't know, something else, and rename `TaskQueueHandle` to `TaskQueue`, making the interface a bit clearer by hiding the difference between a `Queue` and a `Handle` for users.

Questions there:
- Can `spawn_into` be moved onto `TaskQueue`?
- Related: what happens if you try to `spawn_into` with a `LocalExecutor` and a `TaskQueueHandle` that don't match?

Reply:
humm...
(I am thinking ...)

There is some merit but I am not sure I like it.

The first disadvantage  is that now you have many different spawn functions in different entities: if the reader is not careful, one can easily mistake `tq::spawn` with `ex::spawn` and end up doing the wrong thing.

Another consideration is what to do if the `TaskQueue` goes out of scope. It becomes impossible to spawn new tasks into it. Should we then just destroy it?

But last and most importantly, the `TaskQueueHandle` is an opaque that can be passed around (because it is `Copy` to many executors).

Maybe there is a way to make it work but I'd like to see it coupled with a good ergonomic way to use it across executors at least. In any case it would be the subject of a different PR

Reply:
Fair enough; it works well enough as is, and unless a new interface here is clearly better, its not worth it.

> Maybe there is a way to make it work but I'd like to see it coupled with a good ergonomic way to use it across executors at least.

OH, there is something I missed. Is it safe to use the same `TaskQueueHandle` in different executors?

Reply:
Not like this, no.
a `TaskQueue` is registered with a specific executor so if you try to use it with another one it may not exist.

At the end of the day it is just an integer, so it would be useful to have an API to register it in all executors (seastar has such an API, for example)